OpenProblemLibrary/Rochester/setSeries6CompTests/ur_sr_6_10a.pg

```diff
@@ -53,75 +53,55 @@ loadMacros(
-                errors => '',
-                type => 'custom_problem_grader',
-                msg => 'To get full credit, all answers must be correct.  Having all but one correct is worth 50%.  Two or more incorrect answers gives a score of 0%.',
-                                                 );
+    my $evaluated_answers = shift;
+    my $problem_state = shift;
+    my %form_options = @_;
+
+    # initial setup of the answer
+    my $total = 0;
+    my $problem_result = {
+        score => 0,
+        errors => '',
+        type => 'custom_problem_grader',
+        msg => 'To get full credit, all answers must be correct.  Having all but one correct is worth 50%.  Two or more incorrect answers gives a score of 0%.',
+    };
 
     # Return unless answers have been submitted
-    unless ($form_options{answers_submitted} == 1) {
-    
-    # Since this code is in a .pg file we must use double tildes 
-    # instead of Perl's backslash on the next line.
-                return(~~%problem_result,~~%problem_state);
-        }
-        # Answers have been submitted -- process them.
-        
-	########################################################
-	# Here's where we compute the score.  The variable     #
-	# $numright is the number of correct answers.          #
-	########################################################
-
-        my      $numright=0;
-
-        $numright += ($evaluated_answers{'AnSwEr0001'}->{score});
-        $numright += ($evaluated_answers{'AnSwEr0002'}->{score});
-        $numright += ($evaluated_answers{'AnSwEr0003'}->{score});
-        $numright += ($evaluated_answers{'AnSwEr0004'}->{score});
-        $numright += ($evaluated_answers{'AnSwEr0005'}->{score});
-        $numright += ($evaluated_answers{'AnSwEr0006'}->{score});
-
-	if ($numright == 6) {
-		$total = 1;
-	} elsif ($numright == 5) {
-		$total = 0.5;
-	} else {
-		$total = 0;
-	}
-
-        $problem_result{score} = $total; 
-        # increase recorded score if the current score is greater.
-        $problem_state{recorded_score} = $problem_result{score} if $problem_result{score} > $problem_state{recorded_score};
-
-        
-    $problem_state{num_of_correct_ans}++ if $total == 1;
-        $problem_state{num_of_incorrect_ans}++ if $total < 1 ;
-        
-        # Since this code is in a .pg file we must use double tildes 
-    # instead of Perl's backslash on the next line.
-        (~~%problem_result, ~~%problem_state);
+    return($problem_result, $problem_state) unless ($form_options{answers_submitted} == 1);
+
+    # Answers have been submitted -- process them.
+
+    ########################################################
+    # Here's where we compute the score.  The variable     #
+    # $numright is the number of correct answers.          #
+    ########################################################
+    my $numright=0;
+
+    foreach my $key (keys %$evaluated_answers) {
+        # summing the scores is identical to checking each one is correct
```

Comment:
can't you have fractional scores for an answer blank? It's not common but I don't think it's forbidden.  That said the sum of the scores of the answer blank is probably the value you want to test in any case, rather than say, the number of partially correct answers.  I'm pretty sure that the score for an answer blank is expected to be between zero and 1and that requirement is enforced.

Reply:
Yes, other scores are possible -- but for this problem, each response may only be 1 or 0. My comment is specific to this particular problem, and not a general comment about scoring responses.

Reply:
ok.  It makes sense to tune this for a bunch of problems before designing whether there is an over arching principle for designing a custom grader -- possibly one with options.